<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>BAM Processing • TSRexploreR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="BAM Processing">
<meta property="og:description" content="TSRexploreR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">TSRexploreR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BAM_PROCESSING.html">BAM Processing</a>
    </li>
    <li>
      <a href="../articles/DATA_CONDITIONING.html">Data Conditioning</a>
    </li>
    <li>
      <a href="../articles/DIFF_FEATURES.html">Differential Feature Analysis</a>
    </li>
    <li>
      <a href="../articles/FEATURE_SHIFT.html">TSS Shifting</a>
    </li>
    <li>
      <a href="../articles/STANDARD_ANALYSIS.html">Standard Analysis</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="BAM_PROCESSING_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>BAM Processing</h1>
            
      
      
      <div class="hidden name"><code>BAM_PROCESSING.Rmd</code></div>

    </div>

    
    
<p>In addition to accepting TSSs and/or TSRs in a variety of formats, TSRexploreR can import and process alignment BAM files. Any method-specific sequences present in the reads (for instance, the 8-bp UMI and TATAGGG spacer sequence present in the R1 reads of STRIPE-seq datasets) should be trimmed prior to import. Here, we illustrate BAM processing using a small example alignment of paired-end <em>S. cerevisiae</em> <a href="https://genome.cshlp.org/content/30/6/910.long">STRIPE-seq</a> data to the sacCer3 genome assembly.</p>
<div id="prepare-tsrexplorer-object-and-import-bam-file" class="section level2">
<h2 class="hasAnchor">
<a href="#prepare-tsrexplorer-object-and-import-bam-file" class="anchor"></a>Prepare TSRexploreR object and import BAM file</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"TSRexploreR"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org">"ggplot2"</a></span><span class="op">)</span>

<span class="co"># Load example BAM file</span>
<span class="va">bam_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"S288C.bam"</span>, package<span class="op">=</span><span class="st">"TSRexploreR"</span><span class="op">)</span>

<span class="co"># # Load genome assembly</span>
<span class="va">assembly</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"S288C_Assembly.fasta"</span>, package<span class="op">=</span><span class="st">"TSRexploreR"</span><span class="op">)</span>

<span class="co"># Generate sample sheet</span>
<span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>sample_name<span class="op">=</span><span class="st">"S288C"</span>, file_1<span class="op">=</span><span class="va">bam_file</span>, file_2<span class="op">=</span><span class="cn">NA</span><span class="op">)</span>

<span class="co"># Create the TSRexploreR object</span>
<span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tsr_explorer-class.html">tsr_explorer</a></span><span class="op">(</span>sample_sheet<span class="op">=</span><span class="va">samples</span>, genome_assembly<span class="op">=</span><span class="va">assembly</span><span class="op">)</span></code></pre></div>
</div>
<div id="import-bam-file" class="section level2">
<h2 class="hasAnchor">
<a href="#import-bam-file" class="anchor"></a>Import BAM file</h2>
<p>Several filtering options based on BAM flags are available for BAM processing during import. For both single- and paired-end alignments, secondary alignments can be removed. For paired-end datasets (including this example), improperly paired reads and duplicate pairs can be removed.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_bams.html">import_bams</a></span><span class="op">(</span><span class="va">exp</span>, paired<span class="op">=</span><span class="cn">TRUE</span>, proper_pair<span class="op">=</span><span class="cn">TRUE</span>, remove_duplicate<span class="op">=</span><span class="cn">TRUE</span>, remove_secondary<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
</div>
<div id="analysis-of-soft-clipped-bases" class="section level2">
<h2 class="hasAnchor">
<a href="#analysis-of-soft-clipped-bases" class="anchor"></a>Analysis of soft-clipped bases</h2>
<p>A frequent observation in TSS mapping protocols is the presence of spurious, non-templated nucleotides at the 5’ end of the R1 read (<a href="https://www.nature.com/articles/ng1789">Carninci et al. 2006</a>). Addition of a single non-templated G, likely templated by the 7-methylguanosine cap (<a href="https://www.jbc.org/article/S0021-9258(20)30687-6/fulltext">Wulf et al. 2019</a>), is particularly prevalent. If these spurious bases are not incidentally aligned to the genome, they are soft-clipped. TSRexploreR provides two plots for assessment of soft-clipped bases.</p>
<p>First, a plot of the number of reads with certain numbers of soft-clipped bases can be generated. From this, we can see that most reads that were soft-clipped during alignment have a single such base. By default, reads with &gt;3 soft-clipped bases are removed during BAM import.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/softclip_histogram.html">softclip_histogram</a></span><span class="op">(</span><span class="va">exp</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html">scale_fill_viridis_d</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="BAM_PROCESSING_files/figure-html/unnamed-chunk-3-1.png" width="384" style="display: block; margin: auto;"></p>
<p>A plot of base frequency at each soft-clipped position can also be generated. Consistent with previous reports, the majority of soft-clipped bases are G, corresponding to a C on the first-strand cDNA that was presumably templated by the cap.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/softclip_composition.html">softclip_composition</a></span><span class="op">(</span><span class="va">exp</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html">scale_fill_viridis_d</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="BAM_PROCESSING_files/figure-html/unnamed-chunk-4-1.png" width="384" style="display: block; margin: auto;"></p>
</div>
<div id="g-correction" class="section level2">
<h2 class="hasAnchor">
<a href="#g-correction" class="anchor"></a>G correction</h2>
<p>Soft-clipping indicates spurious nucleotide addition at the 5’ end of R1 reads when these bases are not templated to the genome; however, in many cases, such bases may fortuitously match the genome during alignment. Due to the uncertainty of the origin of these bases in reads (reverse transcription of the cap versus a legitimate genomic TSS), TSRexploreR uses a probabilistic approach for their removal. First, the frequency of R1 reads with a soft-clipped G is determined. For each R1 read with a non-soft-clipped 5’ G, a Bernoulli trial is conducted, using the previously determined soft-clipped G frequency as the “success” probability (considered to be the chance that the G will be removed).</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/G_correction.html">G_correction</a></span><span class="op">(</span><span class="va">exp</span><span class="op">)</span></code></pre></div>
</div>
<div id="tss-aggregation" class="section level2">
<h2 class="hasAnchor">
<a href="#tss-aggregation" class="anchor"></a>TSS aggregation</h2>
<p>Following BAM processing, overlapping R1 read 5’ ends are aggregated into TSSs and stored in a GRanges object within the TSRexploreR object at <code>exp@experiment$TSSs</code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tss_aggregate.html">tss_aggregate</a></span><span class="op">(</span><span class="va">exp</span><span class="op">)</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Robert Policastro.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
