<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Data Conditioning â€¢ TSRexploreR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Data Conditioning">
<meta property="og:description" content="TSRexploreR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">TSRexploreR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BAM_PROCESSING.html">BAM Processing</a>
    </li>
    <li>
      <a href="../articles/DATA_CONDITIONING.html">Data Conditioning</a>
    </li>
    <li>
      <a href="../articles/DIFF_FEATURES.html">Differential Feature Analysis</a>
    </li>
    <li>
      <a href="../articles/FEATURE_SHIFT.html">TSS Shifting</a>
    </li>
    <li>
      <a href="../articles/STANDARD_ANALYSIS.html">Standard Analysis</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="DATA_CONDITIONING_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Data Conditioning</h1>
            
      
      
      <div class="hidden name"><code>DATA_CONDITIONING.Rmd</code></div>

    </div>

    
    
<p>When analyzing TSS mapping data, it is often desirable to assess specific subsets of TSSs or TSRs. TSRexploreR offers a flexible approach to data conditioning, allowing grouping by categorical variables and filtering, quantiling, and ordering by numerical variables. Conditionals can be applied to most functions via the <code>data_conditions</code> argument. For heatmaps, the distinct data structure necessitates a separate group of arguments to mimic these data conditioning features. Below, we provide several examples of data conditioning with TSRexploreR, using the provided <em>S. cerevisiae</em> <a href="https://genome.cshlp.org/content/30/6/910.long">STRIPE-seq</a> data.</p>
<div id="prepare-data" class="section level2">
<h2 class="hasAnchor">
<a href="#prepare-data" class="anchor"></a>Prepare data</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"TSRexploreR"</span><span class="op">)</span>

<span class="co"># Load example TSSs</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">TSSs</span><span class="op">)</span>

<span class="co"># Load genome assembly and annotation</span>
<span class="va">assembly</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"S288C_Assembly.fasta"</span>, package<span class="op">=</span><span class="st">"TSRexploreR"</span><span class="op">)</span>
<span class="va">annotation</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"S288C_Annotation.gtf"</span>, package<span class="op">=</span><span class="st">"TSRexploreR"</span><span class="op">)</span>

<span class="co"># Generate sample sheet</span>
<span class="va">sample_sheet</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>
  sample_name<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">"S288C_D_%s"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span>, 
                <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">"S288C_WT_%s"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,
  file_1<span class="op">=</span><span class="cn">NA</span>, file_2<span class="op">=</span><span class="cn">NA</span>,
  condition<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"Diamide"</span>, <span class="fl">3</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"Untreated"</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span>
<span class="op">)</span>

<span class="co"># Create the TSRexploreR object</span>
<span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tsr_explorer-class.html">tsr_explorer</a></span><span class="op">(</span>
  <span class="va">TSSs</span>,
  sample_sheet<span class="op">=</span><span class="va">sample_sheet</span>,
  genome_annotation<span class="op">=</span><span class="va">annotation</span>,
  genome_assembly<span class="op">=</span><span class="va">assembly</span>
<span class="op">)</span>

<span class="co"># Format counts</span>
<span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/format_counts.html">format_counts</a></span><span class="op">(</span><span class="va">exp</span>, data_type<span class="op">=</span><span class="st">"tss"</span><span class="op">)</span>

<span class="co"># Annotate TSSs</span>
<span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/annotate_features.html">annotate_features</a></span><span class="op">(</span><span class="va">exp</span>, data_type<span class="op">=</span><span class="st">"tss"</span>, upstream<span class="op">=</span><span class="fl">250</span>, downstream<span class="op">=</span><span class="fl">100</span>, 
                         feature_type<span class="op">=</span><span class="st">"transcript"</span><span class="op">)</span>

<span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_counts.html">normalize_counts</a></span><span class="op">(</span><span class="va">exp</span>, data_type<span class="op">=</span><span class="st">"tss"</span>, method<span class="op">=</span><span class="st">"deseq2"</span><span class="op">)</span>

<span class="co"># Cluster TSSs</span>
<span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tss_clustering.html">tss_clustering</a></span><span class="op">(</span><span class="va">exp</span>, threshold<span class="op">=</span><span class="fl">3</span>, max_distance<span class="op">=</span><span class="fl">25</span>, n_samples<span class="op">=</span><span class="fl">1</span><span class="op">)</span>

<span class="co"># Associate TSSs with TSRs</span>
<span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/associate_with_tsr.html">associate_with_tsr</a></span><span class="op">(</span><span class="va">exp</span><span class="op">)</span>

<span class="co"># Annotate TSRs</span>
<span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/annotate_features.html">annotate_features</a></span><span class="op">(</span><span class="va">exp</span>, data_type<span class="op">=</span><span class="st">"tsr"</span>, upstream<span class="op">=</span><span class="fl">250</span>, downstream<span class="op">=</span><span class="fl">100</span>, 
                         feature_type<span class="op">=</span><span class="st">"transcript"</span><span class="op">)</span>

<span class="co"># Mark dominant TSS per TSR</span>
<span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mark_dominant.html">mark_dominant</a></span><span class="op">(</span><span class="va">exp</span>, data_type<span class="op">=</span><span class="st">"tss"</span><span class="op">)</span>

<span class="co"># Calculate TSR metrics</span>
<span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tsr_metrics.html">tsr_metrics</a></span><span class="op">(</span><span class="va">exp</span><span class="op">)</span></code></pre></div>
</div>
<div id="grouping-by-a-categorical-variable" class="section level2">
<h2 class="hasAnchor">
<a href="#grouping-by-a-categorical-variable" class="anchor"></a>Grouping by a categorical variable</h2>
<p>In this example, we group TSRs by shape class and plot sequence logos centered on the dominant TSS of each TSR.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_sequence_logo.html">plot_sequence_logo</a></span><span class="op">(</span><span class="va">exp</span>, dominant<span class="op">=</span><span class="cn">TRUE</span>, samples<span class="op">=</span><span class="st">"S288C_WT_1"</span>, 
                   data_conditions<span class="op">=</span><span class="fu"><a href="../reference/conditionals.html">conditionals</a></span><span class="op">(</span>data_grouping<span class="op">=</span><span class="va">shape_class</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="DATA_CONDITIONING_files/figure-html/unnamed-chunk-2-1.png" width="480" style="display: block; margin: auto;"></p>
<p>We also plot IQR width for TSRs split by shape class. As expected, broad TSRs tend to have larger IQRs than peaked TSRs.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_tsr_metric.html">plot_tsr_metric</a></span><span class="op">(</span><span class="va">exp</span>, tsr_metrics<span class="op">=</span><span class="st">"iqr_width"</span>, samples<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"S288C_WT_1"</span><span class="op">)</span>, 
                log2_transform<span class="op">=</span><span class="cn">TRUE</span>, ncol<span class="op">=</span><span class="fl">2</span>, 
                data_conditions<span class="op">=</span><span class="fu"><a href="../reference/conditionals.html">conditionals</a></span><span class="op">(</span>data_grouping<span class="op">=</span><span class="va">shape_class</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">ylim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">8</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="DATA_CONDITIONING_files/figure-html/unnamed-chunk-3-1.png" width="768" style="display: block; margin: auto;"></p>
</div>
<div id="quantiling-by-a-numerical-variable" class="section level2">
<h2 class="hasAnchor">
<a href="#quantiling-by-a-numerical-variable" class="anchor"></a>Quantiling by a numerical variable</h2>
<p>In this example, we split the list of dominant TSSs per TSR into quintiles by score and plot the genomic distribution of TSSs in each quintile. Quantiles are arranged in descending order by default.</p>
<p>By quantiling data by score, we see that the strongest dominant TSSs are more likely to be within the specified promoter window.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_genomic_distribution.html">plot_genomic_distribution</a></span><span class="op">(</span><span class="va">exp</span>, dominant<span class="op">=</span><span class="cn">TRUE</span>, samples<span class="op">=</span><span class="st">"S288C_WT_1"</span>,
                          data_conditions<span class="op">=</span><span class="fu"><a href="../reference/conditionals.html">conditionals</a></span><span class="op">(</span>
                            data_quantiling<span class="op">=</span><span class="fu"><a href="../reference/quantiling.html">quantiling</a></span><span class="op">(</span>by<span class="op">=</span><span class="va">score</span>, n<span class="op">=</span><span class="fl">5</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html">scale_fill_viridis_d</a></span><span class="op">(</span>direction<span class="op">=</span><span class="op">-</span><span class="fl">1</span><span class="op">)</span></code></pre></div>
<p><img src="DATA_CONDITIONING_files/figure-html/unnamed-chunk-4-1.png" width="768" style="display: block; margin: auto;"></p>
</div>
<div id="ordering-by-a-numerical-variable" class="section level2">
<h2 class="hasAnchor">
<a href="#ordering-by-a-numerical-variable" class="anchor"></a>Ordering by a numerical variable</h2>
<p>In this example, we generate a nucleotide color plot around the dominant TSS of each TSR, ordered descending by TSS score.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_sequence_colormap.html">plot_sequence_colormap</a></span><span class="op">(</span><span class="va">exp</span>, dominant<span class="op">=</span><span class="cn">TRUE</span>, samples<span class="op">=</span><span class="st">"S288C_WT_1"</span>,
                       data_conditions<span class="op">=</span><span class="fu"><a href="../reference/conditionals.html">conditionals</a></span><span class="op">(</span>
                         data_ordering<span class="op">=</span><span class="fu"><a href="../reference/ordering.html">ordering</a></span><span class="op">(</span>by<span class="op">=</span><span class="fu">desc</span><span class="op">(</span><span class="va">score</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="DATA_CONDITIONING_files/figure-html/unnamed-chunk-5-1.png" width="288" style="display: block; margin: auto;"></p>
</div>
<div id="heatmap-conditioning" class="section level2">
<h2 class="hasAnchor">
<a href="#heatmap-conditioning" class="anchor"></a>Heatmap conditioning</h2>
<p>As mentioned above, the distinct data structure underlying heatmaps requires a distinct group of functions for application of conditionals. Here, we demonstrate heatmap conditioning using TSRs with significantly different signal between control and diamide-treated samples. We first detect differential TSRs and select those annotated as promoter-proximal. We then generate lists of genes with associated up- or downregulated TSRs and use them to split the heatmaps by the direction of TSR signal change. Heatmaps are also ordered descending by median signal in the control sample.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Build DE model</span>
<span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/fit_de_model.html">fit_de_model</a></span><span class="op">(</span><span class="va">exp</span>, data_type<span class="op">=</span><span class="st">"tsr"</span>, formula<span class="op">=</span><span class="op">~</span><span class="va">condition</span><span class="op">)</span>

<span class="co"># Call differential TSRs</span>
<span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/differential_expression.html">differential_expression</a></span><span class="op">(</span>
  <span class="va">exp</span>, data_type<span class="op">=</span><span class="st">"tsr"</span>,
  comparison_name<span class="op">=</span><span class="st">"Diamide_vs_Untreated"</span>,
  comparison_type<span class="op">=</span><span class="st">"contrast"</span>,
  comparison<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"condition"</span>, <span class="st">"Diamide"</span>, <span class="st">"Untreated"</span><span class="op">)</span>
<span class="op">)</span>

<span class="co"># Annotate differential TSRs</span>
<span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/annotate_features.html">annotate_features</a></span><span class="op">(</span><span class="va">exp</span>, data_type<span class="op">=</span><span class="st">"tsr_diff"</span>, feature_type<span class="op">=</span><span class="st">"transcript"</span><span class="op">)</span>

<span class="co"># Get genes with a promoter-proximal differential TSS</span>
<span class="va">diff_tsrs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/export_for_enrichment.html">export_for_enrichment</a></span><span class="op">(</span><span class="va">exp</span>, data_type<span class="op">=</span><span class="st">"tsr"</span>, keep_unchange<span class="op">=</span><span class="cn">FALSE</span>,
                                   log2fc_cutoff<span class="op">=</span><span class="fl">1</span>, fdr_cutoff<span class="op">=</span><span class="fl">0.05</span>, 
                                   anno_categories<span class="op">=</span><span class="st">"Promoter"</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">transcriptId</span>, <span class="va">de_status</span><span class="op">)</span> 

<span class="co"># Generate named list of differential TSRs</span>
<span class="va">up</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">diff_tsrs</span>, <span class="va">de_status</span><span class="op">==</span><span class="st">"up"</span><span class="op">)</span>
<span class="va">down</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">diff_tsrs</span>, <span class="va">de_status</span><span class="op">==</span><span class="st">"down"</span><span class="op">)</span>

<span class="va">de_tsr_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>up<span class="op">=</span><span class="va">up</span><span class="op">$</span><span class="va">transcriptId</span>, down<span class="op">=</span><span class="va">down</span><span class="op">$</span><span class="va">transcriptId</span><span class="op">)</span>

<span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span><span class="va">exp</span>, data_type<span class="op">=</span><span class="st">"tsr"</span>, samples<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"S288C_WT_1"</span>, <span class="st">"S288C_D_1"</span><span class="op">)</span>, 
             high_color<span class="op">=</span><span class="st">"#440154FF"</span>, downstream<span class="op">=</span><span class="fl">1000</span>, upstream<span class="op">=</span><span class="fl">1000</span>,
             use_normalized<span class="op">=</span><span class="cn">TRUE</span>, order_samples<span class="op">=</span><span class="st">"S288C_WT_1"</span>, 
             split_by<span class="op">=</span><span class="va">de_tsr_genes</span>,  ordering<span class="op">=</span><span class="va">score</span>, order_fun<span class="op">=</span><span class="va">sum</span>, 
             ncol<span class="op">=</span><span class="fl">2</span>, x_axis_breaks<span class="op">=</span><span class="fl">500</span>, rasterize<span class="op">=</span><span class="cn">TRUE</span>, raster_dpi<span class="op">=</span><span class="fl">150</span><span class="op">)</span></code></pre></div>
<p><img src="DATA_CONDITIONING_files/figure-html/unnamed-chunk-6-1.png" width="576" style="display: block; margin: auto;"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Robert Policastro.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
