<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Differential Feature Analysis • TSRexploreR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Differential Feature Analysis">
<meta property="og:description" content="TSRexploreR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">TSRexploreR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BAM_PROCESSING.html">BAM Processing</a>
    </li>
    <li>
      <a href="../articles/DATA_CONDITIONING.html">Data Conditioning</a>
    </li>
    <li>
      <a href="../articles/DIFF_FEATURES.html">Differential Feature Analysis</a>
    </li>
    <li>
      <a href="../articles/FEATURE_SHIFT.html">TSS Shifting</a>
    </li>
    <li>
      <a href="../articles/STANDARD_ANALYSIS.html">Standard Analysis</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="DIFF_FEATURES_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Differential Feature Analysis</h1>
            
      
      
      <div class="hidden name"><code>DIFF_FEATURES.Rmd</code></div>

    </div>

    
    
<p>Changes in gene transcription, driven by alterations in TSS and TSR usage, are pervasive between different environmental conditions, cell types, developmental stages, and normal and disease states. To facilitate differential TSS and TSR analysis, TSRexploreR uses <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8">DESeq2</a> or <a href="https://academic.oup.com/bioinformatics/article/26/1/139/182458">edgeR</a> in an RNA-seq-like analysis, treating TSRs as “genes.”</p>
<div id="preparing-data" class="section level2">
<h2 class="hasAnchor">
<a href="#preparing-data" class="anchor"></a>Preparing Data</h2>
<p>This example uses a set of <a href="https://genome.cshlp.org/content/30/6/910.long">STRIPE-seq</a>-detected TSRs from <em>S. cerevisiae</em> cells treated with 1.5 mM diamide for 1 hour and their corresponding control cells. Diamide is an oxidizing agent that induces substantial changes in transcript levels as measured by microarray analysis (<a href="https://www.molbiolcell.org/doi/10.1091/mbc.11.12.4241">Gasch et al. 2000</a>).</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"TSRexploreR"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org">"ggplot2"</a></span><span class="op">)</span>

<span class="co"># Load example TSSs</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">TSSs</span><span class="op">)</span>

<span class="co"># Load genome annotation</span>
<span class="va">annotation</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"S288C_Annotation.gtf"</span>, package<span class="op">=</span><span class="st">"TSRexploreR"</span><span class="op">)</span>

<span class="co"># Generate sample sheet</span>
<span class="va">sample_sheet</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>
  sample_name<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">"S288C_D_%s"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">"S288C_WT_%s"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,
  file_1<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="cn">NA</span>, <span class="fl">6</span><span class="op">)</span>, file_2<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="cn">NA</span>, <span class="fl">6</span><span class="op">)</span>,
  condition<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"Diamide"</span>, <span class="fl">3</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"Untreated"</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span>
<span class="op">)</span>
<span class="va">sample_sheet</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>
  <span class="va">sample_sheet</span>,
  condition<span class="op">=</span><span class="fu">forcats</span><span class="fu">::</span><span class="fu"><a href="https://forcats.tidyverse.org/reference/fct_relevel.html">fct_relevel</a></span><span class="op">(</span><span class="va">condition</span>, <span class="st">"Untreated"</span><span class="op">)</span>
<span class="op">)</span>

<span class="co"># Create TSRexploreR object</span>
<span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tsr_explorer-class.html">tsr_explorer</a></span><span class="op">(</span><span class="va">TSSs</span>, sample_sheet<span class="op">=</span><span class="va">sample_sheet</span>, genome_annotation<span class="op">=</span><span class="va">annotation</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning in .get_cds_IDX(mcols0$type, mcols0$phase): The "phase" metadata column contains non-NA values for features of type
##   stop_codon. This information was ignored.</code></pre>
</div>
<div id="processing-of-tsss" class="section level2">
<h2 class="hasAnchor">
<a href="#processing-of-tsss" class="anchor"></a>Processing of TSSs</h2>
<p>For expediency, only the TSS processing steps essential to differential TSR analysis are shown here. All additional steps described in the standard analysis vignette can be performed with this data, if desired.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/format_counts.html">format_counts</a></span><span class="op">(</span><span class="va">exp</span>, data_type<span class="op">=</span><span class="st">"tss"</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="../reference/normalize_counts.html">normalize_counts</a></span><span class="op">(</span>data_type<span class="op">=</span><span class="st">"tss"</span>, method<span class="op">=</span><span class="st">"DESeq2"</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="../reference/tss_clustering.html">tss_clustering</a></span><span class="op">(</span>threshold<span class="op">=</span><span class="fl">3</span>, max_distance<span class="op">=</span><span class="fl">25</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning in .local(x, row.names, optional, ...): Arguments in '...' ignored

## Warning in .local(x, row.names, optional, ...): Arguments in '...' ignored

## Warning in .local(x, row.names, optional, ...): Arguments in '...' ignored

## Warning in .local(x, row.names, optional, ...): Arguments in '...' ignored

## Warning in .local(x, row.names, optional, ...): Arguments in '...' ignored

## Warning in .local(x, row.names, optional, ...): Arguments in '...' ignored

## Warning in .local(x, row.names, optional, ...): Arguments in '...' ignored

## Warning in .local(x, row.names, optional, ...): Arguments in '...' ignored

## Warning in .local(x, row.names, optional, ...): Arguments in '...' ignored

## Warning in .local(x, row.names, optional, ...): Arguments in '...' ignored

## Warning in .local(x, row.names, optional, ...): Arguments in '...' ignored

## Warning in .local(x, row.names, optional, ...): Arguments in '...' ignored</code></pre>
</div>
<div id="analysis-of-tsr-similarity" class="section level2">
<h2 class="hasAnchor">
<a href="#analysis-of-tsr-similarity" class="anchor"></a>Analysis of TSR similarity</h2>
<div id="correlation-matrix" class="section level3">
<h3 class="hasAnchor">
<a href="#correlation-matrix" class="anchor"></a>Correlation Matrix</h3>
<p>After normalizing counts and clustering TSSs into TSRs, a correlation matrix can be generated. This allows assessment of replicate similarity and gives a cursory idea of the differences between different biological conditions. As expected, biological triplicates from each condition cluster together, and there is substantially less between-group concordance, consistent with substantial differences in TSR usage.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_correlation.html">plot_correlation</a></span><span class="op">(</span>
  <span class="va">exp</span>, data_type<span class="op">=</span><span class="st">"tsr"</span>, font_size<span class="op">=</span><span class="fl">12</span>,
  use_normalized<span class="op">=</span><span class="cn">TRUE</span>,
  heatmap_colors<span class="op">=</span><span class="fu">viridis</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/viridisLite/man/viridis.html">viridis</a></span><span class="op">(</span><span class="fl">100</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
<p><img src="DIFF_FEATURES_files/figure-html/unnamed-chunk-3-1.png" width="576" style="display: block; margin: auto;"></p>
</div>
<div id="pca" class="section level3">
<h3 class="hasAnchor">
<a href="#pca" class="anchor"></a>PCA</h3>
<p>Another useful way to check differences between samples (and look at replicate concordance) is to perform principal component analysis (PCA). Here, we can see that PC1 nicely separates control and diamide-treated samples.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_reduction.html">plot_reduction</a></span><span class="op">(</span><span class="va">exp</span>, data_type<span class="op">=</span><span class="st">"tsr"</span>, remove_var<span class="op">=</span><span class="fl">0.25</span>, colby<span class="op">=</span><span class="st">"condition"</span><span class="op">)</span></code></pre></div>
<p><img src="DIFF_FEATURES_files/figure-html/unnamed-chunk-4-1.png" width="480" style="display: block; margin: auto;"></p>
</div>
</div>
<div id="differential-tsr-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#differential-tsr-analysis" class="anchor"></a>Differential TSR analysis</h2>
<div id="discovering-differential-tsrs" class="section level3">
<h3 class="hasAnchor">
<a href="#discovering-differential-tsrs" class="anchor"></a>Discovering differential TSRs</h3>
<p>Detection of differential TSRs is performed similarly to differential expression analysis of RNA-seq data with <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8">DESeq2</a> or <a href="https://academic.oup.com/bioinformatics/article/26/1/139/182458">edgeR</a>. First, a DE model containing samples and groups is generated. In this example, we use DESeq2.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/fit_de_model.html">fit_de_model</a></span><span class="op">(</span><span class="va">exp</span>, data_type<span class="op">=</span><span class="st">"tsr"</span>, formula<span class="op">=</span><span class="op">~</span><span class="va">condition</span>, method<span class="op">=</span><span class="st">"DESeq2"</span><span class="op">)</span></code></pre></div>
<p>The model can then be used to call differential TSRs.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/differential_expression.html">differential_expression</a></span><span class="op">(</span>
  <span class="va">exp</span>, data_type <span class="op">=</span> <span class="st">"tsr"</span>, 
  comparison_name <span class="op">=</span> <span class="st">"Diamide_vs_Untreated"</span>,
  comparison_type <span class="op">=</span> <span class="st">"contrast"</span>,
  comparison <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"condition"</span>, <span class="st">"Diamide"</span>, <span class="st">"Untreated"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning in .local(x, row.names, optional, ...): Arguments in '...' ignored</code></pre>
</div>
<div id="annotating-differential-features" class="section level3">
<h3 class="hasAnchor">
<a href="#annotating-differential-features" class="anchor"></a>Annotating Differential Features</h3>
<p>The next step is to annotate the differential features.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/annotate_features.html">annotate_features</a></span><span class="op">(</span><span class="va">exp</span>, data_type<span class="op">=</span><span class="st">"tsr_diff"</span>, feature_type<span class="op">=</span><span class="st">"transcript"</span>, upstream<span class="op">=</span><span class="fl">250</span>, downstream<span class="op">=</span><span class="fl">100</span><span class="op">)</span></code></pre></div>
</div>
<div id="ma-plots" class="section level3">
<h3 class="hasAnchor">
<a href="#ma-plots" class="anchor"></a>MA Plots</h3>
<p>MA plots illustrate the relationship between mean expression (in this case, TSR signal) and fold change.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_ma.html">plot_ma</a></span><span class="op">(</span><span class="va">exp</span>, data_type<span class="op">=</span><span class="st">"tsr"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html">scale_color_viridis_d</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="DIFF_FEATURES_files/figure-html/unnamed-chunk-8-1.png" width="480" style="display: block; margin: auto;"></p>
</div>
<div id="volcano-plots" class="section level3">
<h3 class="hasAnchor">
<a href="#volcano-plots" class="anchor"></a>Volcano Plots</h3>
<p>Volcano plots illustrate the relationship between fold change and significance.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_volcano.html">plot_volcano</a></span><span class="op">(</span><span class="va">exp</span>, data_type<span class="op">=</span><span class="st">"tsr"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html">scale_color_viridis_d</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning: Removed 442 rows containing missing values (geom_point).</code></pre>
<p><img src="DIFF_FEATURES_files/figure-html/unnamed-chunk-9-1.png" width="480" style="display: block; margin: auto;"></p>
</div>
<div id="differential-feature-numbers" class="section level3">
<h3 class="hasAnchor">
<a href="#differential-feature-numbers" class="anchor"></a>Differential feature numbers</h3>
<p>A convenient plot to visualize the number of differential TSRs in the comparisons performed.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_num_de.html">plot_num_de</a></span><span class="op">(</span><span class="va">exp</span>, data_type<span class="op">=</span><span class="st">"tsr"</span>, de_comparisons<span class="op">=</span><span class="st">"Diamide_vs_Untreated"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html">scale_fill_viridis_d</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="DIFF_FEATURES_files/figure-html/unnamed-chunk-10-1.png" width="288" style="display: block; margin: auto;"></p>
</div>
</div>
<div id="gene-ontology" class="section level2">
<h2 class="hasAnchor">
<a href="#gene-ontology" class="anchor"></a>Gene Ontology</h2>
<p>To functionally annotate changes in TSR usage, GO analysis can be performed. TSRexploreR exports annotated differential TSRs ready for GO analysis with <a href="https://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html">clusterProfiler</a>, the package of choice for GO analysis in R. In this example, we export significantly altered TSRs with an annotation of “promoter,” specified above as positions -250 to 100 relative to an annotated gene start. By default, a log2(fold change) cutoff of 1 and an FDR cutoff of 0.05 are enforced. While due to the small size of this example dataset there are no significantly enriched biological process terms, downregulated TSRs are highly associated with genes involved in ribosome biogenesis. This is consistent with the previously reported downregulation of such genes in response to diamide (<a href="https://www.molbiolcell.org/doi/10.1091/mbc.11.12.4241">Gasch et al. 2000</a>, <a href="https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.1001369">Weiner and Chen et al. 2012</a>)</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Export differential promoter TSRs</span>
<span class="va">enrichment_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/export_for_enrichment.html">export_for_enrichment</a></span><span class="op">(</span><span class="va">exp</span>, data_type<span class="op">=</span><span class="st">"tsr"</span>, anno_categories<span class="op">=</span><span class="st">"Promoter"</span>, keep_unchanged<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>

<span class="co"># Perform GO analysis</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">"clusterProfiler"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"org.Sc.sgd.db"</span><span class="op">)</span>

<span class="va">go_enrichment</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/compareCluster.html">compareCluster</a></span><span class="op">(</span>
  <span class="va">geneId</span> <span class="op">~</span> <span class="va">sample</span> <span class="op">+</span> <span class="va">de_status</span>,
  data<span class="op">=</span><span class="va">enrichment_data</span>,
  fun<span class="op">=</span><span class="st">"enrichGO"</span>,
  OrgDb<span class="op">=</span><span class="st">"org.Sc.sgd.db"</span>,
  pAdjustMethod<span class="op">=</span><span class="st">"fdr"</span>,
  ont<span class="op">=</span><span class="st">"BP"</span>,
  keyType<span class="op">=</span><span class="st">"ENSEMBL"</span>
<span class="op">)</span>

<span class="co"># Plot compareCluster results</span>
<span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html">dotplot</a></span><span class="op">(</span><span class="va">go_enrichment</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html">scale_color_viridis_c</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.text.x<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>angle<span class="op">=</span><span class="fl">45</span>, hjust<span class="op">=</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="DIFF_FEATURES_files/figure-html/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;"></p>
</div>
<div id="gene-tracks" class="section level2">
<h2 class="hasAnchor">
<a href="#gene-tracks" class="anchor"></a>Gene Tracks</h2>
<p>Individual examples of differential TSR usage can be viewed as signal tracks. Here, we visualize signal at <em>YDL243</em>/<em>AAD4</em>, encoding an alcohol dehydrogenase involved in the oxidative stress response.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/gene_tracks.html">gene_tracks</a></span><span class="op">(</span>
  <span class="va">exp</span>, feature_name<span class="op">=</span><span class="st">"YDL243C"</span>,
  samples<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>
    TSS<span class="op">=</span><span class="st">"S288C_WT_1"</span>, TSR<span class="op">=</span><span class="st">"S288C_WT_1"</span>,
    TSS<span class="op">=</span><span class="st">"S288C_D_1"</span>, TSR<span class="op">=</span><span class="st">"S288C_D_1"</span>
  <span class="op">)</span>, ymax<span class="op">=</span><span class="fl">35</span>,
  tss_colors<span class="op">=</span><span class="fu">viridis</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/viridisLite/man/viridis.html">viridis</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span>,
  tsr_colors<span class="op">=</span><span class="fu">viridis</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/viridisLite/man/viridis.html">viridis</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
<p><img src="DIFF_FEATURES_files/figure-html/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Robert Policastro.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
