<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>TSS Shifting • TSRexploreR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="TSS Shifting">
<meta property="og:description" content="TSRexploreR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">TSRexploreR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BAM_PROCESSING.html">BAM Processing</a>
    </li>
    <li>
      <a href="../articles/DATA_CONDITIONING.html">Data Conditioning</a>
    </li>
    <li>
      <a href="../articles/DIFF_FEATURES.html">Differential Feature Analysis</a>
    </li>
    <li>
      <a href="../articles/FEATURE_SHIFT.html">TSS Shifting</a>
    </li>
    <li>
      <a href="../articles/STANDARD_ANALYSIS.html">Standard Analysis</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="FEATURE_SHIFT_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>TSS Shifting</h1>
            
      
      
      <div class="hidden name"><code>FEATURE_SHIFT.Rmd</code></div>

    </div>

    
    
<p>Numerous studies have demonstrated that the distribution of TSSs within a given region can shift during development or in response to a given stimulus (<a href="https://www.nature.com/articles/nature12974">Haberle et al. 2014</a>, <a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-017-3834-z">Zhang et al. 2017</a>, <a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-018-4504-5">Danks et al. 2018</a>, <a href="https://www.sciencedirect.com/science/article/pii/S0092867417312400">Ushijima et al. 2018</a>, <a href="https://genome.cshlp.org/content/29/7/1198.full">Lu and Lin 2019</a>). Mathematically speaking, detecting a shift in TSS distribution within a region of interest can be thought of as finding the difference between two discrete probability distributions. To enable detection of TSS shifts, TSRexploreR implements a signed version of <a href="https://en.wikipedia.org/wiki/Earth_mover%27s_distance">earth mover’s distance</a> that we term earth mover’s score (EMS). In essence, we think of the TSS distributions in our two samples to be compared as piles of dirt and ask how much dirt would need to be moved from one pile, how far, and in which direction to recreate the other pile. EMS thus indicates the minimum “cost” of converting one distribution to the other. EMS is between -1 and 1, with larger (or more negative) values indicating larger shifts and the sign indicating direction in accordance with convention for denoting the position of one sequence feature relative to another (negative indicates an upstream shift, positive indicates a downstream shift). EMS is statistically controlled through a permutation test, which produces a p-value and FDR threshold.</p>
<div id="prepare-data" class="section level2">
<h2 class="hasAnchor">
<a href="#prepare-data" class="anchor"></a>Prepare Data</h2>
<p>In this vignette, we detect TSS shifts in TSRs identified by <a href="https://genome.cshlp.org/content/30/6/910.long">STRIPE-seq</a> in control and diamide-treated yeast.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"TSRexploreR"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org">"ggplot2"</a></span><span class="op">)</span>

<span class="co"># Load example TSSs</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">TSSs</span><span class="op">)</span>

<span class="co"># Load genome assembly and annotation</span>
<span class="va">annotation</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"S288C_Annotation.gtf"</span>, package <span class="op">=</span> <span class="st">"TSRexploreR"</span><span class="op">)</span>
<span class="va">assembly</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"S288C_Assembly.fasta"</span>, package <span class="op">=</span> <span class="st">"TSRexploreR"</span><span class="op">)</span>

<span class="co"># Generate sample sheet</span>
<span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>
  sample_name<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">"S288C_D_%s"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">"S288C_WT_%s"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,
  file_1<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="cn">NA</span>, <span class="fl">6</span><span class="op">)</span>, file_2<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="cn">NA</span>, <span class="fl">6</span><span class="op">)</span>,
  condition<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"Diamide"</span>, <span class="fl">3</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"Untreated"</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span>
<span class="op">)</span>

<span class="co"># Create the TSRexploreR object</span>
<span class="va">exp</span> <span class="op">&lt;-</span> <span class="va">TSSs</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="../reference/tsr_explorer-class.html">tsr_explorer</a></span><span class="op">(</span>
    genome_annotation<span class="op">=</span><span class="va">annotation</span>, genome_assembly<span class="op">=</span><span class="va">assembly</span>,
    sample_sheet<span class="op">=</span><span class="va">samples</span>
  <span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="../reference/format_counts.html">format_counts</a></span><span class="op">(</span>data_type <span class="op">=</span> <span class="st">"tss"</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning in .get_cds_IDX(mcols0$type, mcols0$phase): The "phase" metadata column contains non-NA values for features of type
##   stop_codon. This information was ignored.</code></pre>
</div>
<div id="cluster-tsss-and-merge-replicates" class="section level2">
<h2 class="hasAnchor">
<a href="#cluster-tsss-and-merge-replicates" class="anchor"></a>Cluster TSSs and merge replicates</h2>
<p>When detecting TSS shifts, it can be useful to combine replicates. Here, we call TSRs for each sample and then merge TSSs and TSRs from replicates of each condition (3 apiece for untreated and diamide).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tss_clustering.html">tss_clustering</a></span><span class="op">(</span><span class="va">exp</span>, threshold<span class="op">=</span><span class="fl">3</span>, n_samples<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span>  
  <span class="fu"><a href="../reference/merge_samples.html">merge_samples</a></span><span class="op">(</span>data_type <span class="op">=</span> <span class="st">"tss"</span>, merge_group<span class="op">=</span><span class="st">"condition"</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="../reference/merge_samples.html">merge_samples</a></span><span class="op">(</span>data_type <span class="op">=</span> <span class="st">"tsr"</span>, merge_group<span class="op">=</span><span class="st">"condition"</span><span class="op">)</span></code></pre></div>
</div>
<div id="calculate-shifting" class="section level2">
<h2 class="hasAnchor">
<a href="#calculate-shifting" class="anchor"></a>Calculate Shifting</h2>
<p>We now calculate shift scores for a set of consensus TSRs. In this approach, merged TSRs from the untreated and diamide-treated conditions are themselves merged if they are within 100 bp of one another and both have at least 10 counts. By default, an FDR threshold of 0.05 is used to select significantly shifted TSRs</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tss_shift.html">tss_shift</a></span><span class="op">(</span>
  <span class="va">exp</span>,
  sample_1<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>TSS<span class="op">=</span><span class="st">"Untreated"</span>, TSR<span class="op">=</span><span class="st">"Untreated"</span><span class="op">)</span>,
  sample_2<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>TSS<span class="op">=</span><span class="st">"Diamide"</span>, TSR<span class="op">=</span><span class="st">"Diamide"</span><span class="op">)</span>,
  comparison_name<span class="op">=</span><span class="st">"Untreated_vs_Diamide"</span>,
  max_distance <span class="op">=</span> <span class="fl">100</span>, min_threshold <span class="op">=</span> <span class="fl">10</span>, n_resamples <span class="op">=</span> <span class="fl">1000L</span>
<span class="op">)</span></code></pre></div>
</div>
<div id="shifting-plots" class="section level2">
<h2 class="hasAnchor">
<a href="#shifting-plots" class="anchor"></a>Shifting Plots</h2>
<p>As an initial assessment of the results, a plot of shift scores for significantly shifted regions can be generated.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_shift_rank.html">plot_shift_rank</a></span><span class="op">(</span><span class="va">exp</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html">scale_fill_viridis_c</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_hline</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></code></pre></div>
<p><img src="FEATURE_SHIFT_files/figure-html/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;"></p>
<p>A stacked barplot of the number of upstream and downstream shifts can be also generated.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_shift_count.html">plot_shift_count</a></span><span class="op">(</span><span class="va">exp</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html">scale_fill_viridis_d</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="FEATURE_SHIFT_files/figure-html/unnamed-chunk-5-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="annotate-shifts" class="section level2">
<h2 class="hasAnchor">
<a href="#annotate-shifts" class="anchor"></a>Annotate Shifts</h2>
<p>To facilitate further analysis of the shifted regions, they can be annotated relative to known features.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/annotate_features.html">annotate_features</a></span><span class="op">(</span><span class="va">exp</span>, data_type<span class="op">=</span><span class="st">"shift"</span>, feature_type<span class="op">=</span><span class="st">"transcript"</span><span class="op">)</span></code></pre></div>
<p>The genomic distribution of shifts can then be plotted.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_genomic_distribution.html">plot_genomic_distribution</a></span><span class="op">(</span><span class="va">exp</span>, data_type<span class="op">=</span><span class="st">"shift"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html">scale_fill_viridis_d</a></span><span class="op">(</span>direction<span class="op">=</span><span class="op">-</span><span class="fl">1</span><span class="op">)</span></code></pre></div>
<p><img src="FEATURE_SHIFT_files/figure-html/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;"></p>
</div>
<div id="gene-tracks" class="section level2">
<h2 class="hasAnchor">
<a href="#gene-tracks" class="anchor"></a>Gene Tracks</h2>
<p>For visual assessment of shift results, signal tracks can be visualized at a region of interest. For reference, annotated shifts are stored in <code>exp@shifting$results$Untreated_vs_Diamide</code> for this example.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/gene_tracks.html">gene_tracks</a></span><span class="op">(</span>
  <span class="va">exp</span>, feature_name<span class="op">=</span><span class="st">"YDR418W"</span>,
  samples<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>
    TSS<span class="op">=</span><span class="st">"Untreated"</span>, TSR<span class="op">=</span><span class="st">"Untreated"</span>,
    TSS<span class="op">=</span><span class="st">"Diamide"</span>, TSR<span class="op">=</span><span class="st">"Diamide"</span>
  <span class="op">)</span>, promoter_only<span class="op">=</span><span class="cn">TRUE</span>,
  tss_colors<span class="op">=</span><span class="fu">viridis</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/viridisLite/man/viridis.html">viridis</a></span><span class="op">(</span><span class="fl">2</span>, end<span class="op">=</span><span class="fl">0.5</span><span class="op">)</span>,
  tsr_colors<span class="op">=</span><span class="fu">viridis</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/viridisLite/man/viridis.html">viridis</a></span><span class="op">(</span><span class="fl">2</span>, end<span class="op">=</span><span class="fl">0.5</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
<p><img src="FEATURE_SHIFT_files/figure-html/unnamed-chunk-8-1.png" width="768" style="display: block; margin: auto;"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Robert Policastro.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
